cirq/contrib/svg/svg.py

```diff
@@ -60,45 +62,104 @@ def _fit_horizontal(tdd: 'cirq.TextDiagramDrawer',
     return col_starts, col_widths
 
 
+def _fit_vertical(tdd: 'cirq.TextDiagramDrawer',
+                  ref_boxheight: float, row_padding: float) \
+        -> Tuple[List[float], List[float], Dict[float, int]]:
+    # Note: y values come as half integers. Map to integers
+    all_yis = sorted({yi for _, yi in tdd.entries.keys()}
+                     | {yi1 for _, yi1, _, _ in tdd.vertical_lines}
+                     | {yi2 for _, _, yi2, _ in tdd.vertical_lines}
+                     | {yi for yi, _, _, _ in tdd.horizontal_lines})
+    yi_map = {yi: i for i, yi in enumerate(all_yis)}
+
+    max_yi = max(yi_map[yi] for yi in all_yis)
+    row_heights = [0.0] * (max_yi + 2)
+    for (_, yi), v in tdd.entries.items():
+        yi = yi_map[yi]
+        row_heights[yi] = max(ref_boxheight, row_heights[yi])
+
+    for yi in all_yis:
+        row_heights[yi_map[yi]] += row_padding
+
+    row_starts = [0.0]
+    for i in range(1, max_yi + 3):
+        row_starts.append(row_starts[i - 1] + row_heights[i - 1])
+
+    return row_starts, row_heights, yi_map
+
+
+def _debug_spacing(col_starts, row_starts):
```

Comment:
Type annotations.

What's the "_debug" for?

Reply:
added docstring and note

Reply:
Not going to add type annotations here since the variable names give you all the info you need (i.e. you should only ever pass variables named those exact things)